tests/wake-format/basic/stdout

```diff
@@ -982,3 +981,61 @@ export def !(x: Boolean): Boolean =
     if x then False else True
 
 
+export def mapPartial (f: a => Option b): (list: List a) => List b =
+    def loop = match _
+        Nil = Nil
+        h, t =
+            # don't wait on f to process tail:
+            def sub = loop t
+            match (f h)
+                Some x = x, sub
+                None = sub
+    loop
+
+
+target writeImp inputs mode path content =
+    def writeRunner =
+        def imp m p c = prim "write"
+        def pre input = Pair input Unit
+        def post = match _
+            Pair (Fail f) _ = Fail f
+            Pair (Pass output) Unit =
+                if mode < 0 || mode > 0x1ff then
+                    Fail (makeError "write {path}: Invalid mode ({strOctal mode})")
+                else match (imp mode path content)
+                    Fail f = Fail (makeError f)
+                    Pass path = Pass (editRunnerOutputOutputs (path, _) output)
```

Comment:
Are these the constructors the main PR was targeting?  I don't see much else other than the initial `emccFn (` which seems to be exercised, though I've also not looked at the C++ to know *what* to look for here.  Nevertheless, looks good!

Reply:
No, I think the test here was the "else match" part. Though these constructors not getting exploded does show a bit of that functionality